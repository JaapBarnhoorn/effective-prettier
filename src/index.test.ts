import { promises as fs } from "fs"

import prettier from "."

jest.setTimeout(30000)

async function testFormat(filePath) {
  const input = await fs.readFile(filePath, { encoding: "utf-8" })
  const output = await prettier.formatText(input, filePath, {
    verbose: true,
    skipIgnore: true
  })
  return output
}

describe("Auto formatting", () => {
  test("JavaScript 1", async () => {
    expect(await testFormat("./src/fixtures/javascript-1.js")).toMatchSnapshot()
  })

  test("JavaScript 2", async () => {
    expect(await testFormat("./src/fixtures/javascript-2.js")).toMatchSnapshot()
  })

  test("TypeScript 1", async () => {
    expect(await testFormat("./src/fixtures/typescript-1.ts")).toMatchSnapshot()
  })

  test("JSON 1", async () => {
    expect(await testFormat("./src/fixtures/json-1.json")).toMatchSnapshot()
  })

  test("Markdown 1", async () => {
    expect(await testFormat("./src/fixtures/markdown-1.md")).toMatchSnapshot()
  })

  test("CSS in JS - JSS", async () => {
    expect(await testFormat("./src/fixtures/cssinjs-jss.jsx")).toMatchSnapshot()
  })

  test("CSS in JS - Styled Components", async () => {
    expect(await testFormat("./src/fixtures/cssinjs-styled.jsx")).toMatchSnapshot()
  })

  test("HTML with CSS 1", async () => {
    expect(await testFormat("./src/fixtures/htmlcss-1.html")).toMatchSnapshot()
  })
})
