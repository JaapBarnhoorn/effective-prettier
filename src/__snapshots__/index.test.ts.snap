// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auto formatting Autofix 1 1`] = `
"function hello() {
  const message = \\"powerful\\"
  return message
}
"
`;

exports[`Auto formatting Autofix 2 1`] = `
"const msgs = [ \\"hello\\", \\"world\\", \\"from\\", \\"our\\", \\"test\\" ]
"
`;
